### DISCUSSION TITLE
Forbid renovate to update a PR if a specific label is present?

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [renovatebot/renovate#27470] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.186.1

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### What is your question?

We use Renovate on some repos that uses an in-house merge queue.
We've had some issues in that after a renovate PR enters the merge queue it updates the PR because a new version has been released.
That complicates things a bit since the PR won't end up being closed since there's new changes in there etc.
It likely is something we need to improve in our merge queue, since it feels like it should handle that.

However I think it would be useful to be able to have something like a label or something to instruct renovate to not touch this PR again for whatever reason, until forced to rebase or the label is removed. 

Does that make sense at all?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by HonkingGoose** [comment]
## Package rule: `rebaseWhen=never` for special label?

We have a `rebaseWhen=never` option, maybe you can use that with a special label in your repository? I'm thinking of something like this:

1. Create special label `specialLabel`
1. Create Renovate packageRule with `rebaseWhen=never` for the `specialLabel`

But maybe that's not going to work properly.

## `rebaseLabel` requests rebases...

Renovate has a `rebaseLabel` config option, but that's to _request_ a rebase, not avoid a rebase!

Maybe we need a `rebaseIgnoreLabels` feature?

## Relevant docs

I did find some more docs related to your problem:

- [Renovate docs, Updating and rebasing branches](https://docs.renovatebot.com/updating-rebasing/)
- [Renovate docs, `rebaseWhen` config option](https://docs.renovatebot.com/configuration-options/#rebasewhen)
- [Renovate docs, `rebaseLabel`](https://docs.renovatebot.com/configuration-options/#rebaselabel)

## Docs need updating

A quick skim of the FAQ and a basic search does not really tell me if we have something to help you. So once we figure out how to "forbid Renovate to rebase when special label", we should put it in the [Renovate docs, Updating and rebasing branches](https://docs.renovatebot.com/updating-rebasing/), and other places. 😄

**Reply by HenrikPoulsen** [reply]
For the packageRules approach I haven't seen that it supports basically a `matchLabels`. If that existed then yeah maybe using that with `rebaseWhen=never` might work. Though from the docs [here](https://docs.renovatebot.com/updating-rebasing/#newer-dependency-versions) it hints that if a new version is detected then rebaseWhen is ignored and it will update regardless. Is that true?
If so then I don't think there's any mechanism today to forbid renovate to update a PR/branch whatever you configure it to, if there's a new version available


`rebaseIgnoreLabel` as you mentioned would also work if it had existed (and assuming it would also ignore new version updates while that label is set).

At least from what I can see it doesn't seem like this scenario is supported right now.
Is this something that would make sense for renovate to offer?

**Reply by HonkingGoose** [reply]
I'll let the maintainers answer your questions, because I don't know enough to help further. 🙃

**Comment by rarkins** [comment, marked as answer]
Perhaps a `stopUpdatingLabel` configuration, similar to `rebaseLabel`

**Reply by HenrikPoulsen** [reply]
🤦 thanks. Completely missed that. Will that be respected when there's a new version detected as well?
The docs are a bit ambiguous on that point

**Reply by rarkins** [reply]
Yes I think that was the intention of this feature. If you can confirm that's how it behaves then feel free to PR the docs to remove any ambiguity

**Reply by HonkingGoose** [reply]
@rarkins Do you mean make a _new_ `stopUpdatingLabel` config option, or do you mean a specific user configuration that uses the existing `rebaseLabel` thing? Right now it's a bit confusing for me. :smile:

**Reply by rarkins** [reply]
When I wrote that I'd forgotten that we already supported stopUpdatingLabel!

**Reply by HonkingGoose** [reply]
Oh we already _have_ the `stopUpdatingLabel`... :smile:
